src/systems/dynamic_detachable_joint/DynamicDetachableJoint.cc

```diff
@@ -0,0 +1,386 @@
+    _res.set_message("Invalid request: child_model_name and child_link_name must be set.");
+    return true;
+  }
+
+  if (_req.command().empty())
+  {
+    _res.set_success(false);
+    _res.set_message("Invalid request: command must be 'attach' or 'detach'.");
+    return true;
+  }
+
+   // If attach is requested
+   if (_req.command() == "attach")
+   {
+     if (this->isAttached) 
+     {
+       _res.set_success(false);
+      _res.set_message("Already attached to child model " + this->attachedChildModelName +
+                       " at link " + this->attachedChildLinkName + ".");
+       gzdbg << "Already attached" << std::endl;
+       return true;
+     }
+
+    // set the child model and link names from the request
+    this->childModelName = _req.child_model_name();
+    this->childLinkName  = _req.child_link_name();
+     this->OnAttachRequest(msgs::Empty());
+     _res.set_success(true);
+     _res.set_message("Attached to child model " + this->childModelName +
+                      " at link " + this->childLinkName + ".");
+   }
+
+   // If detach is requested
+   else if (_req.command() == "detach")
+   {
+     if (!this->isAttached)
+     {
+         _res.set_success(false);
+         _res.set_message(std::string("Detach request received for ")
+              + this->attachedChildModelName + "/" + this->attachedChildLinkName);
+         gzdbg << "Already detached" << std::endl;
+         return true;
+     }
+
+    // Validate that the request matches what is actually attached.
+    const auto &reqModel = _req.child_model_name();
+    const auto &reqLink  = _req.child_link_name();
+    if (reqModel != this->attachedChildModelName ||
+        reqLink  != this->attachedChildLinkName)
+    {
+      _res.set_success(false);
+      _res.set_message(
+        "Detach rejected: requested " + reqModel + "/" + reqLink +
+        " but currently attached to " + this->attachedChildModelName + "/" +
+        this->attachedChildLinkName + "."
+      );
+      return true;
+    }
+
+     this->OnDetachRequest(msgs::Empty());
+     _res.set_success(true);
+     _res.set_message("Detached from child model " + this->attachedChildModelName +
+                      " at link " + this->attachedChildLinkName + ".");
+   }
+
+   else
+   {
+     _res.set_success(false);
+     _res.set_message("Invalid command. Use 'attach' or 'detach'.");
+     return true;
+   }
+   return true;
+ }
+
+//////////////////////////////////////////////////
+void DynamicDetachableJoint::OnAttachRequest(const msgs::Empty &)
+{
+  this->attachRequested = true;
+}
+
+//////////////////////////////////////////////////
+void DynamicDetachableJoint::OnDetachRequest(const msgs::Empty &)
+{
+  this->detachRequested = true;
+}
+
+//////////////////////////////////////////////////
+void DynamicDetachableJoint::PublishJointState(bool attached)
+{
+  gz::msgs::StringMsg detachedStateMsg;
+  if (attached)
+  {
+    detachedStateMsg.set_data("attached to " +
```

Comment:
Why are we publishing a custom message with custom semantics? I'm not sure we should publish this at all.

Reply:
`DetachableJoint.cc` publishes only 2 types of outputs: "attached" or "detached", in your case you publish "attached to ..." (if I were redisigning this I'd just publish a bool). If I were writing a parser to interact with your software I'd have to look up this magic string. Its much better to either publish an empty string or just the linked name so end users dont have to write custom parsers. Alternatively publish the entity/knullentity that the object is attached to.